packages/manager/apps/pci-block-storage/src/pages/attach/AttachStorage.page.tsx

```diff
@@ -21,22 +21,28 @@ export default function AttachStorage() {
   const { projectId, volumeId } = useParams();
   const { t } = useTranslation('attach');
   const { addError, addSuccess } = useNotifications();
-  const { data: volume } = useVolume(projectId, volumeId);
+  const { data: volume, isPending: isVolumePending } = useVolume(
+    projectId,
+    volumeId,
+  );
   const { data: dataInstances, isPending: isInstancesPending } = useInstances(
     projectId,
     volume?.region,
   );
-  const instances = useMemo(
-    () =>
-      !!dataInstances &&
+  const instances = useMemo(() => {
```

Comment:
This could be a `select` function (like `filterInstancesByVolumeAvailability`) given to the `useInstances` hook with `volume` as parameter.